### PR TITLE
feat: C509 Implement type `Time` and `~biguint`

### PR DESCRIPTION
**Pull request by bkioshn**
# Description

Implement encode and decode for type `Time` and `~biguint`

## Related Issue(s)

#608 

## Description of Changes

- Implement encode and decode for type `Time` and `~biguint` and provide test cases

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst-gateway-crates/c509-certificate/src/c509_time.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        assert_eq!(decoded_time, time);
+    }
+
+    // Test reference https://datatracker.ietf.org/doc/draft-ietf-cose-cbor-encoded-cert/09/
+    // A.1.  Example RFC 7925 profiled X.509 Certificate
+    #[test]
+    fn test_encode_decode() {
+        let mut buffer = Vec::new();
+        let mut encoder = minicbor::Encoder::new(&mut buffer);
+        let time = Time::new(1_672_531_200);
```
Comment (apskhem) [inline catalyst-gateway-crates/c509-certificate/src/c509_time.rs:80]: Make the number as a constant.

File: catalyst-gateway-crates/c509-certificate/src/c509_big_uint.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
+//! C509 Unwrapped CBOR Unsigned Bignum
+
+// cspell: words Bignum bignum
+
+use minicbor::{encode::Write, Decode, Decoder, Encode, Encoder};
+
+/// A struct representing an unwrapped CBOR unsigned bignum.
+#[derive(Debug, Clone, PartialEq)]
+pub struct UnwrappedBigUint(u64);
```
Comment (apskhem) [inline catalyst-gateway-crates/c509-certificate/src/c509_big_uint.rs:9]: Should also put the detail about why it's called "unwrapped" here in the docs comment.

File: catalyst-gateway-crates/c509-certificate/src/c509_big_uint.rs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        Ok(())
+    }
+}
+
+impl Decode<'_, ()> for UnwrappedBigUint {
+    fn decode(d: &mut Decoder<'_>, _ctx: &mut ()) -> Result<Self, minicbor::decode::Error> {
+        let bytes = d.bytes()?;
+        // Turn bytes into u64
+        let b = bytes.iter().fold(0, |acc, &b| (acc << 8) | u64::from(b));
+        Ok(UnwrappedBigUint::new(b))
```
Comment (apskhem) [inline catalyst-gateway-crates/c509-certificate/src/c509_big_uint.rs:41]:
```suggestion
        let b = d
            .bytes()?
            .iter()
            .fold(0, |acc, &b| (acc << 8) | u64::from(b));
        Ok(UnwrappedBigUint::new(b))
```

File: catalyst-gateway-crates/c509-certificate/src/c509_time.rs (hunk 1 above)
Comment (bkioshn) [reply catalyst-gateway-crates/c509-certificate/src/c509_time.rs:80]: Will add a comment above instead, it is just a number I took from the example for testing purpose. @apskhem

File: catalyst-gateway-crates/c509-certificate/src/c509_big_uint.rs (hunk 2 above)
Comment (bkioshn) [reply catalyst-gateway-crates/c509-certificate/src/c509_big_uint.rs:9]: Will point to a document about CDDL unwrap.

**Review by stevenj — APPROVED** [review]
Looks good to me

**Review by apskhem — APPROVED** [review]

**Pull request merged**